activerecord/CHANGELOG.md

```diff
@@ -3,6 +3,18 @@
 
     *Jason Nochlin*
 
+*   Fix an issue in the `where.associated` method only allows filtering associated records with an `inner join`,
+    which can potentially override existing join types in the query.
+    The fix ensures that the association is joined using the appropriate join type (either inner join or left outer join) based on the existing joins in the scope.
+    This prevents unintentional overrides of existing join types and ensures consistency in the generated SQL queries.
+    Example:
+    asoiciated will use left_joins alternative use deafult joins
+    ```ruby
+        Post.left_joins(:author).where.associated(:author)
+    ```
```

Comment:
```suggestion
    
    ```ruby
    # associated will use LEFT JOIN for this query instead of using JOIN
    Post.left_joins(:author).where.associated(:author)
    ```
```